### ISSUE TITLE
[Windows] Missing keyboard navigation code

### ISSUE THREAD

**Issue by tim-weis** [rust-windowing/winit#1590]
*winit* currently doesn't support keyboard navigation in windows (e.g. <kbd>TAB</kbd> navigation). Windows' window manager implements this already (see [IsDialogMessageW](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-isdialogmessagew)), but *winit* doesn't employ it.

It would be great to get support for keyboard navigation in *winit* out of the box. To get there, all message-retrieval code would need to be revisited and updated to either:
* include a call to `IsDialogMessageW`, or
* arrange for a user-supplied callback, so users can opt-in if they desire.

I acknowledge that neither of these suggestions is trivial to implement, with the latter likely opening up a whole new can of worms.

The following links by [Raymond Chen](https://devblogs.microsoft.com/oldnewthing/) may prove to be helpful in evaluating cost and risk for implementing the former:

* [Using the TAB key to navigate in non-dialogs](https://devblogs.microsoft.com/oldnewthing/20031021-00/?p=42083)
* [Using the TAB key to navigate in non-dialogs, redux](https://devblogs.microsoft.com/oldnewthing/20131009-00/?p=2983)

**Comment by Osspial** [comment]
I don't think it's Winit's place to do this, since there isn't a clear simple API for the feature and (as you said) it's non-trivial to implement. However, once #1508 is merged it should be possible to implement this outside of Winit's codebase.
